src/server/controllers/checkPasswordToken.ts

```diff
@@ -12,63 +12,133 @@ import { renderer } from '@/server/lib/renderer';
 import { logger } from '@/server/lib/logger';
 import { PasswordRoutePath } from '@/shared/model/Routes';
 import { PasswordPageTitle } from '@/shared/model/PageTitle';
+import { getConfiguration } from '@/server/lib/getConfiguration';
+import { authenticate as authenticateWithOkta } from '@/server/lib/okta/api/authentication';
+
+const { okta } = getConfiguration();
 
 export const checkPasswordTokenController = (
   path: PasswordRoutePath,
   pageTitle: PasswordPageTitle,
 ) =>
   handleAsyncErrors(async (req: Request, res: ResponseWithRequestState) => {
-    let requestState = res.locals;
-    const { token } = req.params;
+    const { useOkta } = res.locals.queryParams;
+    if (okta.registrationEnabled && useOkta && path === '/welcome') {
```

Comment:
Maybe this controller is now getting a bit long with these extra okta changes added. Could it be worth moving this ```if (okta.registrationEnabled && useOkta && path === '/welcome') { ... } else { ... }``` logic up into the `routes/setPassword.tx` and make a seperate Okta only controller file ? Then this file can just be dedicated to Identity stuff?

Reply:
That's funny, because the first time I started adding to this controller I ended up doing a refactoring PR which broke it up from one controller into two and moved some code out into libs (https://github.com/guardian/gateway/pull/1275), so it's a lot smaller than it used to be! I can refactor it further if you think that it's still too big

Reply:
Personally I think having them in the same file is less error-prone because you can see all the logic in one place and if something needs to be changed in both scenarios, it is less likely to be missed

Reply:
👍